js/index.js

```diff
@@ -1,51 +1,38 @@
-// Get the html container element
-const htmlContainerElement = document.getElementById('mainContent');
-
-// Modal elements
+// Getting DOM elements
 const modal = document.getElementById('myModal');
 const modalContent = document.getElementById('modalContent');
 const input = document.getElementById('searchInput');
+const htmlContainerElement = document.getElementById('mainContent');
 
-// Request/Response
-
-let httpRequest;
-let responseContent;
-
-// Received data storage
+// Global scope variables
 
 let parsed;
+let offset = 0;
+const limit = 35; //number of gifs to be displayed at once
+const step = limit;
+let pageContent = '';
+let search = '';
 
-input.addEventListener('keyup', event => {
-  //used destructuring for event.keyCode => event object is destructured and {keyCode} takes it's value.
-  if (event.keyCode === 13) {
-    const search = event.target.value;
-    searchRequest(search);
-  }
-});
+//Functions
 
 const onLoad = () => {
   trendingRequest();
-
-  window.onclick = ({ target }) => {
-    //used destructuring for event.target => event object is destructured and {target} takes it's value.
-    if (target == modalContent) {
-      modalContent.innerHTML = '';
-      modal.style.display = 'none';
-    }
-  };
 };
 
-const trendingRequest = () => {
+const trendingRequest = extend => {
   // API URL Request data
   const baseURL = 'https://api.giphy.com';
   const apiEndpoint = '/v1/gifs/trending';
   const apiKey = '?api_key=hHGFaUaGZMVAaFuK6GaC5yUk3ceeykET';
```

Comment:
Why you didn't declare these consts in separate variables since same values are used at line 40-43?

Reply:
apiEndpoint is not the same but line 24 and line 26 I can declare in separate variables but it will flood the global scope. However, if it is better, I'll do it.

Reply:
Then I can recommend using common base endpoint url (`const GIFS_ENDPOINT = '/v1/gifs'`) and use it for different paths (such as `const TRENDING = '${GIFS_ENDPOINT}/trending'` and same for search or whatever else exists on that base endpoint)

Reply:
That's a common approach and best practice when you have a base for any endpoint you have and have to call it with different parameters. Also you just should always check for your code duplicates and remove them, to keep your code clean and beautiful

Reply:
fixed